extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -1,32 +1,38 @@
-import {Layout_Enum, applyFillContent} from '#core/dom/layout';
+import * as Preact from '#core/dom/jsx';
+import {Layout_Enum} from '#core/dom/layout';
 
-import {AmpStoryPageAttachment} from 'extensions/amp-story/1.0/amp-story-page-attachment';
-
-const TAG = 'amp-story-shopping-attachment';
-
-export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
+export class AmpStoryShoppingAttachment extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
-    /** @private {string} */
-    this.myText_ = TAG;
-
-    /** @private {?Element} */
-    this.container_ = null;
+    /** @private @return {!Element} */
+    this.attachmentEl_ = null;
   }
 
   /** @override */
   buildCallback() {
-    super.buildCallback();
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    applyFillContent(this.container_, /* replacedContent */ true);
+    this.attachmentEl_ = (
+      <amp-story-page-attachment
+        layout="nodisplay"
+        theme={this.element.getAttribute('theme')}
```

Comment:
Will the `theme` attribute be mandatory for `amp-story-shopping-attachment`? If not, then this line could end up as `theme=null` or `theme=`.

Reply:
With the methods in jsx.js the attribute won't be set if it's null, check it out:
https://github.com/ampproject/amphtml/blob/main/src/core/dom/jsx.js#L66

Reply:
That's cool, thanks for showing me!